### PR TITLE
OptionValue docs

### PR DESCRIPTION
**Pull request by kaapstorm**
## Technical Summary

Adds some documentation to the `optionvalue` module, and removes an unused class.

### Safety story

No change to functionality.

### Automated test coverage

Includes a doctest.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/motech/repeaters/optionvalue.py [hunk 1]
```diff
@@ -1,4 +1,48 @@
-import attr
+"""
+The OptionValue property
+========================
+
+``OptionValue`` is a subclass of ``property``. It is used for managing
+the values in a dictionary named "options" belonging to the object that
+the OptionValue is a property of.
```
Comment (millerdev) [inline corehq/motech/repeaters/optionvalue.py:7]:
```suggestion
the ``OptionValue`` is a property of.
```

File: corehq/motech/repeaters/optionvalue.py [hunk 2]
```diff
@@ -1,4 +1,48 @@
...
+the following ``Meal`` class to manage meals on an airline::
+
+    >>> @define
+    ... class Meal:
+    ...     options = field(factory=dict)
+    ...     dish = OptionValue()
+    ...     category = OptionValue(choices=["chicken", "beef", "vegan"])
+
+The ``dish`` and ``category`` OptionValues will manage values in the
+``options`` dictionary named ``dish`` and ``category`` respectively.
```
Comment (millerdev) [inline corehq/motech/repeaters/optionvalue.py:19]:
```suggestion
``options`` dictionary with keys ``'dish'`` and ``'category'`` respectively.
```

File: corehq/motech/repeaters/optionvalue.py [hunk 3]
```diff
@@ -1,4 +1,48 @@
...
+properties on Django models and their subclasses to be stored in a
+JSON field. You can find examples in ``SQLCaseRepeater``, which has
+options "version", "white_listed_case_types" and "black_listed_users",
+and its subclass ``SQLDhis2EntityRepeater``, which has an additional
+option, "dhis2_entity_config". Values for all of these options are
+persisted by their base class, ``SQLRepeater.options``, defined as
+``JSONField(default=dict)``.
+
+"""
+from attrs import define, field  # noqa  # Required by docstring above
```
Comment (millerdev) [inline corehq/motech/repeaters/optionvalue.py:45]: This is weird to put code in the module that is only needed for the docstring. That doesn't seem quite right. I can think of some possible alternatives:
- Inject these values into the module before running doctests
  ```py
  def test_doctests():
      import corehq.motech.repeaters.optionvalue as mod
      from attrs import define, field
      mod.define = define
      mod.field = field
      results = doctest.testmod(mod)
      assert results.failed == 0
- Add the import to the docstring
  ```py
  >>> from attrs import define, field
  >>> @define
  ...
  ```

**Comment by millerdev** [comment]
Thanks for writing these docs!

**Review by millerdev — APPROVED** [review]

**Review by AmitPhulera — APPROVED** [review]
Thanks, @kaapstorm for writing these ❤️

**Pull request merged**